### ISSUE TITLE
Android O: Picture-in-Picture support

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/focus-android#820]

**Comment by pocmo** [comment]
Tried this today. Works nicely.

![pip-focus](https://user-images.githubusercontent.com/89638/30924142-330fc4c2-a3ae-11e7-90d3-6bdcd7730b56.png)

**Comment by bbinto** [comment]
@pocmo is this already working? What work is left to do?

**Comment by pocmo** [comment]
> @pocmo is this already working? What work is left to do?

No. The screenshot above is from a hack prototype to try whether that would be possible at all.

I assume we only want to do that for video content that is playing in full-screen. We know whether we are in full-screen in GeckoView - however we do not know if a video is playing. Briefly talked to the GV team and they were thinking about adding some media APIs that would allow us to do that.

**Comment by pocmo** [comment]
For the record: PiP support comes with Android 8+ (which seem to be surprisingly ~30% of our users)

**Comment by jonalmeida** [comment]
We have [`PictureInPictureFeature`](https://github.com/mozilla-mobile/android-components/blob/main/components/feature/session/src/main/java/mozilla/components/feature/session/PictureInPictureFeature.kt) in Components that we can add to Focus quite easily too.
